### ISSUE TITLE
test for get_model_path() fails with withr 2.3.0

### ISSUE THREAD

**Issue by copernican** [metrumresearchgroup/rbabylon#286]
From the [changelog](https://withr.r-lib.org/news/index.html#withr-2-3-0-2020-09-22) for withr 2.3.0:

> `defer()` now throws an error if an error occurs in the deferred expression

As a result, the test `get_model_path() works no paths found` fails. `create_temp_model()` will attempt to delete the model file it creates when the test context exits, but this file will not exist because that test manually deletes it. Previously, this error was being swallowed.

```r
library(fs)
library(testthat)
library(withr)

packageVersion("withr")
#> [1] '2.3.0'
f <- function(envir = parent.frame()) {
  temp <- tempfile()
  writeLines("foo", temp)
  defer(file_delete(temp), envir)
  temp
} 
test_that("all good", {
  x <- f()
  expect_equal(readLines(x), "foo")
})
#> Test passed 🥇
test_that("throw error", {
  x <- f()
  expect_equal(readLines(x), "foo")
  file_delete(x)
})
#> ── ERROR (???): throw error ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
#> Error: [ENOENT] Failed to remove '/tmp/RtmpfbBic4/file62c06ec1b93b': no such file or directory
#> Backtrace:
```

**Comment by seth127** [comment]
potentially fixed in https://github.com/metrumresearchgroup/rbabylon/pull/289 but we should verify

**Comment by seth127** [comment]
Closed by #289
